src/ControllerV2/RosalieInteroperabilityController.php

```diff
@@ -14,47 +14,55 @@
 {
+    public function __construct(
+        private readonly RosalieService $service,
+        private readonly EntityManagerInterface $em,
+    ) {
+    }
+
     #[IsGranted('ROLE_MEMBRE')]
     #[Route('/beneficiaries/{id}/add-si-siao-number', name: 'add_si_siao_number')]
-    public function addSiSiaoNumber(Request $request, Beneficiaire $beneficiary, RosalieService $service, TranslatorInterface $translator, EntityManagerInterface $em): Response
+    public function addSiSiaoNumber(Request $request, Beneficiaire $beneficiary): Response
     {
         $beneficiaryCreationProcess = $beneficiary->getCreationProcess();
 
         $redirection = $beneficiaryCreationProcess?->getIsCreating()
             ? $this->redirectToRoute('create_beneficiary', ['id' => $beneficiaryCreationProcess->getId(), 'step' => $beneficiaryCreationProcess->getLastReachedStep()])
             : $this->redirectToRoute('list_beneficiaries');
 
-        return $this->processSiSiaoNumberForm($request, $beneficiary, $service, $translator, $em, $redirection);
+        return $this->processSiSiaoNumberForm($request, $beneficiary, $redirection);
     }
 
     #[IsGranted('ROLE_MEMBRE')]
     #[Route('/beneficiary/{id}/affiliate/add-si-siao-number', name: 'affiliate_beneficiary_add_si_siao_number')]
-    public function affiliationAddSiSiaoNumber(Request $request, Beneficiaire $beneficiary, RosalieService $service, TranslatorInterface $translator, EntityManagerInterface $em): Response
+    public function affiliationAddSiSiaoNumber(Request $request, Beneficiaire $beneficiary): Response
     {
-        return $this->processSiSiaoNumberForm($request, $beneficiary, $service, $translator, $em, $this->redirectToRoute('affiliate_beneficiary_relays', ['id' => $beneficiary->getId()]));
+        return $this->processSiSiaoNumberForm($request, $beneficiary, $this->redirectToRoute('affiliate_beneficiary_relays', ['id' => $beneficiary->getId()]));
     }
 
-    private function processSiSiaoNumberForm(Request $request, Beneficiaire $beneficiary, RosalieService $service, TranslatorInterface $translator, EntityManagerInterface $em, RedirectResponse $redirection): Response
+    private function processSiSiaoNumberForm(Request $request, Beneficiaire $beneficiary, RedirectResponse $redirection): Response
     {
         $form = $this->createFormBuilder()
             ->add('number', TextType::class, ['label' => 'si_siao_number', 'data' => $beneficiary->getSiSiaoNumber()])
             ->getForm()->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
+            $beneficiaryCheck = $this->service->checkBeneficiaryOnRosalie($beneficiary);
             $beneficiary->setSiSiaoNumber($form->get('number')->getData());
-            if ($service->beneficiaryExistsOnRosalie($beneficiary)) {
-                $service->linkBeneficiaryToRosalie($beneficiary);
-                $this->addFlash('success', $translator->trans('si_siao_number_found_rosalie'));
+
```

Comment:
I think the flush should be there anyway to commit the sisiao number, because we also call flush in `linkBeneficiaryToRosalie` below, what do you think ?